builder/dockerfile/internals_test.go

```diff
@@ -211,89 +211,113 @@ othergrp:x:6666:
 
 	// positive tests
 	for _, testcase := range []struct {
+		builder   *Builder
 		name      string
 		chownStr  string
-		idMapping *idtools.IDMappings
-		expected  idtools.IDPair
+		idMapping *idtools.IdentityMapping
+		state     *dispatchState
+		expected  idtools.Identity
 	}{
 		{
+			builder:   &Builder{options: &types.ImageBuildOptions{Platform: "linux"}},
 			name:      "UIDNoMap",
 			chownStr:  "1",
 			idMapping: unmapped,
-			expected:  idtools.IDPair{UID: 1, GID: 1},
+			state:     &dispatchState{},
```

Comment:
If state is the same value for every case there's no reason to add it to the struct. Just create it in the for loop.